### PR TITLE
[stable/minio] Add support for securityContext runAsUser and runAsGroup 

### LINKED ISSUE #14596: [stable/minio] securityContext / runasuser
As containers/pods should run without root permissions if possible, securityContext/RunAsUser should be implemented, and default to a minio user (e.g. 1001)

Bitnami helm chart already has this for minio, but that lacks other features from this chart.

Example implementation:
For standalaone: https://github.com/bitnami/charts/blob/f37e9a84d6bc0f9ce0ad6f6fb0bb2b217ec911c4/bitnami/minio/templates/deployment-standalone.yaml#L58 and for distributed: https://github.com/bitnami/charts/blob/f37e9a84d6bc0f9ce0ad6f6fb0bb2b217ec911c4/bitnami/minio/templates/statefulset.yaml#L68

As minio already can run as non-root, this should be possible to implement

### LINKED ISSUE #17370: [stable/minio] securityContext is not supported
**Is your feature request related to a problem? Please describe.**
Same problem as https://github.com/helm/charts/issues/14596.

We are running this chart with a security policy of no root users.

**Describe the solution you'd like**
The minio chart to support the securityContext parameter and set the user and group of a deployment when the chart is rendered.

**Describe alternatives you've considered**
We have used the bitnami chart to deploy minio for the moment, but would like this to be available in the official repository.

**Additional context**

We will be forking the project and providing a pull request shortly.

### PR DESCRIPTION
**Pull request by simonwgill**
FAO @Acaleph, @minio
#### What this PR does / why we need it:
When the podSecurityContext does not allow runAsRoot, there was no method to configure minio to function. This PR allows consumers to use minio in this situation.

#### Is this a new chart?

No

#### Which issue this PR fixes

  - fixes #17370
  - fixes #14596

#### Special notes for your reviewer:

The configPath and configPathmc variables have been changed to use a default that doesn't rely on the active user being root.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by simonwgill** [comment]
Should I be keeping track of the head branch, or should I wait for some level of approval of the branch as it was?

**Comment by nitisht** [comment]
Please fix the conflicts @simonwgill

**Comment by nitisht** [comment]
can you confirm if you have deployed this locally and it works @simonwgill

**Comment by simonwgill** [comment]
@nitisht I had to go back and double check, but yes, it's happily running locally on my setup.

**Comment by nitisht** [comment]
Thank you for checking again! :)

File: stable/minio/README.md [hunk 1]
```diff
@@ -167,6 +167,10 @@ The following table lists the configurable parameters of the MinIO chart and the
...
 | `metrics.serviceMonitor.interval`         | Scrape interval. If not set, the Prometheus default scrape interval is used                                                             | `nil`                                      |
 | `metrics.serviceMonitor.scrapeTimeout`    | Scrape timeout. If not set, the Prometheus default scrape timeout is used                                                               | `nil`                                      |
+| `securityContext.enabled`    | Will allow changing the UID or GID from root. If true, one of securityContext.runAsUser or securityContext.runAsGroup must be defined                                                             | `false`                                      |
+
+| `securityContext.runAsUser`    | The user id to run the minio process as within the pod                                                              | `1000`                                      |
+| `securityContext.runAsUser`    | The group id to run the minio process as within the pod                                                              | `nil`                                      |
```
Comment (nitisht) [inline stable/minio/README.md]: should be `securityContext.runAsGroup`

File: stable/minio/README.md [hunk 2]
```diff
@@ -108,8 +108,8 @@ The following table lists the configurable parameters of the MinIO chart and the
...
 | `accessKey`                               | Default access key (5 to 20 characters)                                                                                                 | `AKIAIOSFODNN7EXAMPLE`                     |
 | `secretKey`                               | Default secret key (8 to 40 characters)                                                                                                 | `wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY` |
-| `configPath`                              | Default config file location                                                                                                            | `~/.minio`                                 |
-| `configPathmc`                            | Default config file location for MinIO client - mc                                                                                      | `~/.mc`                                    |
+| `configPath`                              | Default config file location                                                                                                            | `/etc/minio`                                 |
```
Comment (nitisht) [inline stable/minio/README.md]: `--config-dir` has been deprecated from MinIO server as of minio/minio#7033, can you please remove this option completely?

File: stable/minio/values.yaml [hunk 3]
```diff
@@ -48,11 +48,18 @@ priorityClassName: ""
 existingSecret: ""
 accessKey: "AKIAIOSFODNN7EXAMPLE"
 secretKey: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
-configPath: "/root/.minio/"
-configPathmc: "/root/.mc/"
+configPath: "/etc/minio/"
```
Comment (nitisht) [inline stable/minio/values.yaml]: `--config-dir` has been deprecated from MinIO server as of minio/minio#7033, can you please remove this option completely?

**Review by nitisht — CHANGES_REQUESTED** [review]
Few recommendations..

File: stable/minio/README.md (hunk 2 above)
Comment (simonwgill) [reply stable/minio/README.md]: I've removed the option. Haven't implemented the certs-dir functionality at this point though.

Are you expecting --certs-dir to be implemented under this ticket? That seems out of scope for what I was updating and I'm not sure how to set up any tests to handle it. Doesn't look like anyone raised the deprecation as an issue either.

**Comment by simonwgill** [comment]
Does the --cert-dir option need support adding in this PR? That seems out of scope for the original issue.

**Comment by dmccaffery** [comment]
Just ran into this myself -- what do we need to do to get the ball rolling @nitisht / @simonwgill ? -- looks like there are a couple of conflicts -- is anything else required?

**Comment by simonwgill** [comment]
I wasn't sure if adding --cert-dir was in scope for this PR, and I've been waiting on an answer to that question from someone. 

Haven't looked at what's been integrated since I last rebased though.

**Pull request closed without merging**